asv_bench/benchmarks/groupby.py

```diff
@@ -149,39 +113,40 @@ class Nth(object):
 
     goal_time = 0.2
 
-    def setup_cache(self):
```

Comment:
Not sure if there is a reason why we are intermixing `setup_cache` and `setup` within this module. The former seems preferable given these setups aren't super resource intensive and when refactoring this to use `setup` I didn't encounter any issues locally. May be worth a separate cleanup to consistently use `setup` throughout the module

Reply:
that's fine

Reply:
While refactoring the benchmarks, I mainly preferred `setup_cache` over `setup` to avoid running the setup n times for each method of the benchmarking class. 

I think it should be okay to change them to just `setup`, just need to monitor that it will not significantly increase the time of the ASV CI run on travis. (which I don't think it will)

Reply:
Thanks @mroeschke that makes sense. Should be easy to revert if it causes an issue